### ISSUE TITLE
Default to https when protocol is file

### ISSUE THREAD

**Issue by hailiang-wang** [mapbox/mapbox.js#866]
hi, guys

I get a problem from with L.marker . I use mapbox.js within a [apache cordova](http://cordova.apache.org) project which is for iOS.  cordova is a hybrid developing platform for mobile apps.

I add mapbox.js into the web related pages. After adding below codes into my cordova/www/index.html, I can get the map which is very cool.

```
        L.mapbox.accessToken = 'pk.eyJ1Ijoi';
        var map = L.mapbox.map('map', 'hain1ci75')
        .setView([0, 50], 3);
        // add a marker in the given location, attach some popup content to it and open the popup
        L.marker([-31, -12]).addTo(map)
            .bindPopup('A pretty CSS3 popup. <br> Easily customizable.')
            .openPopup();
        }
```

You can see, I also add a marker, but it fails to render the image.
From the iOS simulator and its debug console, I find something about these two images.

![image](https://cloud.githubusercontent.com/assets/3538629/4148855/7958156e-342c-11e4-8f2f-a2c8718c4bbf.png)

I think their paths are not well computed which should be 'js/lib/mapbox/images/marker-icon-2x.png' or 'http://api.tiles.mapbox.com/mapbox.js/v2.0.1/images/marker-icon-2x.png'

![image](https://cloud.githubusercontent.com/assets/3538629/4148835/22fdbdf4-342c-11e4-8583-73ac1951c5e5.png)

So, I think this maybe a bug.

**Comment by hailiang-wang** [comment]
hi, folks 

I am able to fix this bug by modifying a line in src/mapbox.js . 
In line 5, it is 

```
     'use strict';
    // Hardcode image path, because Leaflet's autodetection
    // fails, because mapbox.js is not named leaflet.js
    **window.L.Icon.Default.imagePath = '//api.tiles.mapbox.com/mapbox.js/' + 'v' +**
    require('../package.json').version + '/images';
```

If I change the imagePath to 'http://api.tiles.mapbo ... , nice marker shows up !
But I am not sure the impact of this change. Does anyone can give some suggestions ?

Thanks.

**Comment by tmcw** [comment]
Yeah, so we [fix this with protocols](https://github.com/mapbox/mapbox.js/blob/a02a9d00b1545884b74a095f8ee97228c2cd72c3/src/request.js#L12) but need to do so with our base URLs... Cordova/PG use `file://` urls which trip up our relative protocol technique.

**Comment by tmcw** [comment]
Fixed in a857e507d83c1ae517a83f65e7f06d0c8c903872 - this will be included in the next release.
